### PR TITLE
Notebook detection fix

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

File: wandb/lib/ipython.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
-        if 'terminal' in get_ipython().__module__:
-            return 'ipython'
-        else:
-            return 'jupyter'
-    except (NameError, AttributeError):
+        from IPython import get_ipython
+    except ImportError:
         return "python"
+
+    if get_ipython() is None:
```
Comment (vanpelt) [inline wandb/lib/ipython.py:7]: Don't you mean `get_ipython` ?

Comment (farizrahman4u) [reply wandb/lib/ipython.py:7]: Nope. We are testing the return value of the function.

Comment (vanpelt) [reply wandb/lib/ipython.py:15]: But `get_ipython` can be None in the `ImportError` case...

Comment (farizrahman4u) [reply wandb/lib/ipython.py:6]: In the `ImportError` case we return, so rest of the code is not executed.

Comment (vanpelt) [reply wandb/lib/ipython.py:6]: Duh, cool 👍

**Review by raubitsj — APPROVED** [review]

**Pull request merged**